File: tests/detectors/samples/S02/python/negative.py
import random, requests, time

BASE_S = 0.1
MAX_S = 20.0

def backoff_s(attempt):
    capped = min(BASE_S * 2 ** attempt, MAX_S)
    return capped * (0.5 + random.random() * 0.5)

def call_with_retry(url):
    for attempt in range(5):
        r = requests.get(url, timeout=5)
        if r.ok:
            return r
        time.sleep(backoff_s(attempt))
    raise RuntimeError("give up")

File: tests/detectors/samples/S02/typescript/negative.ts
const BASE_MS = 100;
const MAX_MS = 20_000;

function backoffMs(attempt: number): number {
  const capped = Math.min(BASE_MS * Math.pow(2, attempt), MAX_MS);
  return Math.round(capped * (0.5 + Math.random() * 0.5));
}

export async function callWithRetry(url: string) {
  for (let attempt = 0; attempt < 5; attempt++) {
    const res = await fetch(url, { signal: AbortSignal.timeout(5_000) });
    if (res.ok) return res;
    await new Promise(r => setTimeout(r, backoffMs(attempt)));
  }
  throw new Error("give up");
}
